spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestMerge.java

```diff
@@ -186,11 +189,14 @@ public void testMergeWithStaticPredicatePushDown() {
             row(2L, "hardware") // new
             );
     assertEquals(
-        "Output should match", expectedRows, sql("SELECT * FROM %s ORDER BY id, dep", tableName));
+        "Output should match",
+        expectedRows,
+        sql("SELECT * FROM %s ORDER BY id, dep", selectTarget()));
   }
 
   @Test
   public void testMergeIntoEmptyTargetInsertAllNonMatchingRows() {
+    Assume.assumeTrue("Branch can only be used for non-empty table", branch == null);
```

Comment:
I think that we should fix this so that at least `main` works. Otherwise, `main` doesn't work but omitting branch does, which is weird because those are supposed to be the same thing (omitting scans main).

Reply:
I got this working for `main` using the change suggested in https://github.com/apache/iceberg/pull/6965/files#r1125768603 and this update to `useRef`:

```java
  public ThisT useRef(String name) {
    Preconditions.checkArgument(
        snapshotId() == null, "Cannot override ref, already set snapshot id=%s", snapshotId());
    Snapshot snapshot = table().snapshot(name);
    Preconditions.checkArgument(snapshot != null || name.equals("main"), "Cannot find ref %s", name);
    if (snapshot != null) {
      return newRefinedScan(table(), tableSchema(), context().useSnapshotId(snapshot.snapshotId()));
    } else {
      return newRefinedScan(table(), tableSchema(), context());
    }
  }
```

Reply:
Thanks for the suggestion an verification! Once we can have the experience discussion settled I will come back to fix this part.